src/manifold/src/manifold.cpp

```diff
@@ -71,6 +73,67 @@ Manifold Halfspace(Box bBox, glm::vec3 normal, float originOffset) {
   float zDeg = glm::degrees(glm::atan(normal.y, normal.x));
   return cutter.Rotate(0.0f, yDeg, zDeg);
 }
+
+double DistanceToVector(const glm::vec3& pt, const glm::vec3& v) {
+  return glm::length(pt - v * glm::dot(pt, v));
+}
+
+inline double DistanceToPlane(const glm::vec3& pt, const glm::vec4& plane) {
+  return glm::dot({plane.x, plane.y, plane.z}, pt) - plane.w;
+}
+
+glm::vec4 PlaneOfPts(const glm::vec3& a, const glm::vec3& b,
+                     const glm::vec3& c) {
+  auto crx = glm::cross(c - a, b - a);
+  auto n = glm::length(crx);
+  if (n == 0.) return glm::vec4();
+  return glm::vec4(crx, glm::dot(crx, a)) / n;
+}
+
+// Return a trio of non-collinear indices into pts. If none are found, indices
+// will be -1 (and hulling cannot be performed).
+glm::ivec3 NonCollinearTriple(const std::vector<glm::vec3>& pts,
+                              const double precision) {
+  const int len = pts.size();
+  int furthest = 1;
```

Comment:
Just wondering, do we need to find the furthest point or any non-collinear triple is fine? If we don't need the furthest point, we can probably just try each point repeatedly until some non-collinear triple is found, which should be faster than iterating through all the points and calculating their distances.

Reply:
Yeah I guess picking *good* triangles will be important for the algorithm. However, picking the furthest point may result in a sliver triangles as well.

Reply:
I should say, it's more about the plane that the triangle forms, so if it's a sliver it doesn't really matter. So long as it cuts across the shape I think. In any case, this is very much not where the performance needs to be improved.